zyngine/zynthian_engine_pianoteq_jsonrpc.py

```diff
@@ -0,0 +1,696 @@
+				value = ElementTree.Element('VALUE')
+				value.set('name', 'audio-setup')
+				devicesetup = ElementTree.SubElement(value, 'DEVICESETUP')
+				devicesetup.set('deviceType', 'JACK')
+				devicesetup.set('audioOutputDeviceName', 'Auto-connect OFF')
+				devicesetup.set('audioInputDeviceName', 'Auto-connect OFF')
+				devicesetup.set('audioDeviceRate', str(PIANOTEQ_SAMPLERATE))
+				devicesetup.set('forceStereo', '0')
+				root.append(value)
+
+			if midi_setup_node:
+				logging.debug("Fixing MIDI Setup ")
+				for midisetup in midi_setup_node.iter('midi-setup'):
+					midisetup.set('listen-all', '0')
+			else:
+				logging.debug("Creating new MIDI Setup")
+				value = ElementTree.Element('VALUE')
+				value.set('name', 'midi-setup')
+				midisetup = ElementTree.SubElement(value, 'midi-setup')
+				midisetup.set('listen-all', '0')
+				root.append(value)
+
+			if crash_node is not None:
+				if crash_node.attrib['val']:
+					logging.warning("Pianoteq detected previous crash ({})".format(crash_node.attrib['val']))
+					crash_node.attrib['val'] = ''
+
+			tree.write(PIANOTEQ_CONFIG_FILE)
+
+		except Exception as e:
+			logging.error("Fixing Pianoteq config failed: {}".format(e))
+			return format(e)
+
+
+# ------------------------------------------------------------------------------
+# Pianoteq module constants & parameter configuration/initialization
+# ------------------------------------------------------------------------------
+
+PIANOTEQ_SW_DIR = os.environ.get('ZYNTHIAN_SW_DIR', "/zynthian/zynthian-sw") + "/pianoteq6"
+PIANOTEQ_BINARY = PIANOTEQ_SW_DIR + "/pianoteq"
+
+PIANOTEQ_CONFIG_DIR = os.path.expanduser("~") + "/.config/Modartt"
+PIANOTEQ_DATA_DIR = os.path.expanduser("~") + '/.local/share/Modartt/Pianoteq'
+PIANOTEQ_ADDON_DIR = PIANOTEQ_DATA_DIR + '/Addons'
+PIANOTEQ_MY_PRESETS_DIR = PIANOTEQ_DATA_DIR + '/Presets'
+PIANOTEQ_MIDIMAPPINGS_DIR = PIANOTEQ_DATA_DIR + '/MidiMappings'
+
+try:
+	PIANOTEQ_VERSION = list(map(int, os.environ.get('PIANOTEQ_VERSION').split(".")))
+	PIANOTEQ_PRODUCT = os.environ.get('PIANOTEQ_PRODUCT')
+	PIANOTEQ_TRIAL = int(os.environ.get('PIANOTEQ_TRIAL'))
+except:
+	info = get_pianoteq_binary_info()
+	if info:
+		PIANOTEQ_VERSION = list(map(int, str(info['version']).split(".")))
+		PIANOTEQ_PRODUCT = str(info['product'])
+		PIANOTEQ_TRIAL = int(info['trial'])
+	else:
+		PIANOTEQ_VERSION = [6, 5, 1]
+		PIANOTEQ_PRODUCT = "STAGE"
+		PIANOTEQ_TRIAL = 1
+
+PIANOTEQ_NAME = "Pianoteq{}{}".format(PIANOTEQ_VERSION[0], PIANOTEQ_VERSION[1])
+
+if PIANOTEQ_VERSION[0] > 6 or (PIANOTEQ_VERSION[0] == 6 and PIANOTEQ_VERSION[1] >= 5):
+	PIANOTEQ_JACK_PORT_NAME = "Pianoteq"
+else:
+	PIANOTEQ_JACK_PORT_NAME = PIANOTEQ_NAME
+
+if PIANOTEQ_PRODUCT != "STAGE":
+	PIANOTEQ_CONFIG_FILENAME = "{}.prefs".format(PIANOTEQ_NAME)
+else:
+	PIANOTEQ_CONFIG_FILENAME = "{} {}.prefs".format(PIANOTEQ_NAME, PIANOTEQ_PRODUCT)
+
+PIANOTEQ_CONFIG_FILE = PIANOTEQ_CONFIG_DIR + "/" + PIANOTEQ_CONFIG_FILENAME
+
+
```

Comment:
Could/should we split these auxiliary functions so we have a single/common copy of them for both implementations?

Reply:
Yes we should! I will look at this.

Reply:
I have moved all the common code to "pianoteq.py". Some functions are static and called from "pianoteq6.py". I decided not to subclass Pianoteq as it proved too challenging. It was also too challenging to consolidate to a single class, hence there are still two classes: pianoteq and pianteq6. The latter is for pianoteq versions < 7.5.